### PR TITLE
Add support for getting the Google auth tokens using Google's service account

### PR DESCRIPTION
**Pull request by antas-marcin**
### What's being changed:

This PR adds a possibility of using Google Authorization SDK to automatically obtaining the token.

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: usecases/modulecomponents/apikey/google.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+		if apiKey := modulecomponents.GetValueFromContext(ctx, "X-Google-Studio-Api-Key"); apiKey != "" {
+			return apiKey, nil
+		}
+	} else {
+		if apiKey := modulecomponents.GetValueFromContext(ctx, "X-Google-Vertex-Api-Key"); apiKey != "" {
+			return apiKey, nil
+		}
+		if useGoogleAuth {
+			// This method checks all possible places for Google credentials and if successful gets the token source
+			tokenSource, err := google.DefaultTokenSource(ctx)
```
Comment (dirkkul) [inline usecases/modulecomponents/apikey/google.go:34]: how exactly would this work? A user adds some kind of token (which?) as a header to a request and the library automatically extracts it?

Comment (antas-marcin) [reply usecases/modulecomponents/apikey/google.go:34]: This is the same approach as AWS auth SDK has. The Google Auth SDK checks standard places and looks if some service account has been configured if yes then uses it to authorize in GCP:

```
// It looks for credentials in the following places,
// preferring the first location found:
//
//  1. A JSON file whose path is specified by the
//     GOOGLE_APPLICATION_CREDENTIALS environment variable.
//     For workload identity federation, refer to
//     https://cloud.google.com/iam/docs/how-to#using-workload-identity-federation on
//     how to generate the JSON configuration file for on-prem/non-Google cloud
//     platforms.
//  2. A JSON file in a location known to the gcloud command-line tool.
//     On Windows, this is %APPDATA%/gcloud/application_default_credentials.json.
//     On other systems, $HOME/.config/gcloud/application_default_credentials.json.
//  3. On Google App Engine standard first generation runtimes (<= Go 1.9) it uses
//     the appengine.AccessToken function.
//  4. On Google Compute Engine, Google App Engine standard second generation runtimes
//     (>= Go 1.11), and Google App Engine flexible environment, it fetches
//     credentials from the metadata server.
```

Comment (dirkkul) [reply usecases/modulecomponents/apikey/google.go:34]: oki, so let's say this runs on WCD - what would I need to do?

Comment (antas-marcin) [reply usecases/modulecomponents/apikey/google.go:34]: The simplest way would be to provide `GOOGLE_APPLICATION_CREDENTIALS` during deployment, a better would be to configure K8s IAM Service Account to provide those creds directly into our pods

**Review by dirkkul — DISMISSED** [review]

**Review by dirkkul — APPROVED** [review]

**Pull request merged**